.travis.yml

```diff
@@ -0,0 +1,162 @@
+  - secure: "nSSkp37NCAvwlj1WUKIqm8alH+yPsNA6efbGAYO1weclSr+ZK/QWoSp3oMEM0QvWHska5X4Xosdn3G8rRbSzaD/ZkFnkOVpyfWjJlPZbgKRZn07N0koRnq2sNp3HKYlc+2lOd8iIkFq52AcJQOhdVo0DsFlPwrFPj50zUCyML9vObvJa4WSzA7esTD4xArUpbajkYYJvTPViM/JlY8h5I0//lm1oauIj6c+w7vfTTEEpXoohp0QBgs6F54WjyQG/Pn5ARedDv1+F7FXutaFvcaHvMgHB44ft87QP0mGDcdjMYnAgzfgvSVykqzzXgR0a7qyUfHBeVvUbSxT0aRG8pgynvHKrfO12nyv6VyTPUAyv4PYPt0wcsZfyDYQgRvz7fH8+ZDK2XnyTdF32xSTbMmTepAIhrdBbvoecBuogBUOG/yjuUYF1CR7xxWU6q5vYSNh33U1zIM/3skcdeh+tiqbCP6TguBPbPE9jYqI0O9EVl9XE9btuVaN0VS1ELBJW22wW8rDRqQIt+onWwLSf1iS2HyU+Qj3TL3rIz3/i8oxPrLAMwpswGxI4bvXYz27LVfr6DDF8w5s0+tDmC0uzS3L7ZmhlokoVOUWFu5u4ETg69eEA/UXiVppnZArZ9Efg3uiobuFZIbaGXjCNn1EmX5GWu6eddaHgYmiwRXwoyQM="
+  - DOCKER_NAMED_CONTAINER=test_extractor   # Used to name the container for easier finding
+  - TEST_CLOWDER_USERNAME=test@example.com
+  - TEST_CLOWDER_PASSWORD=testPassword
+  - TEST_SPACE_NAME=test_space
+
+# Basic setup for the build. Install software needed by all extractors followed by optional installs for specific extractors
+before_install:
+  - echo "Working with job:\ $EXTRACTOR_BRANCH"
+  - echo "Checking branch name against folder structure"
+  - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
+  - echo "Travis branch name is:\ $BRANCH"
+  - BUILD_FOLDER=`./test/check_branch_acceptable.sh "$BRANCH" "$EXTRACTOR_BRANCH"`
+  - echo "Verified build folder is ${BUILD_FOLDER} for branch ${BRANCH}"
+  - echo "Displaying basic environment information"
+  - docker --version
+  - ls -l
+  - ls -l test
+  - echo "Attempting to update to docker-compose version:\ $DOCKER_COMPOSE_VERSION"
+  - sudo rm -f /usr/local/bin/docker-compose   # Update docker-compose by removing any old versions
+  - sudo curl -L "https://github.com/docker/compose/releases/download/1.24.0/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+  - sudo chmod +x /usr/local/bin/docker-compose
+  - ls -l /usr/local/bin/docker-compose     # Show that docker-compose is in place with correct permissions
+  - docker-compose --version
+  # Install common packages
+  - echo "Installing packages"
+  - pip install -U pyclowder
+  # Extractor dependency installations
+  - sudo apt install -y libjpeg-dev
+  - sudo apt install -y libtiff-dev
+  - pip install -U opencv-python
+  - pip install -U numpy
+
+# Start up the test bed containers and copy files from other locations as needed
+install:
+  - echo "Starting up environment with docker-compose"
+  - docker-compose -p clowder -f test/docker-compose.yml up -d
+  - docker ps -a
+  - docker inspect clowder_clowder_1
+  - ./test/wait_for_clowder.sh
+  # Setup the clowder account
+  - echo "Starting Clowder configuration"
+  - echo "Clowder URL:\ $CLOWDER_HOST_URI"
+  - curl -X POST -H "Content-Type:application/x-www-form-urlencoded" "$CLOWDER_HOST_URI/signup?email=$TEST_CLOWDER_USERNAME"
+  - ./test/wait_for_registration.sh
```

Comment:
this can be done using the clowder/mongo-init container see https://opensource.ncsa.illinois.edu/bitbucket/projects/CATS/repos/clowder/browse/scripts/mongo-init

Reply:
This approach isn't working due to social.users not existing in MongoDB when it's newly downloaded.

Reply:
not true, it should create the collection. I'll check to make sure it works.

Reply:
Some more information:
- docker-compose -p clowder up -d
- docker ps
```CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS                            PORTS                                                                                         NAMES
8490603dabda        clowder/clowder:latest       "/bin/sh -c /home/cl…"   8 seconds ago       Up 4 seconds (health: starting)   0.0.0.0:9000->9000/tcp                                                                        clowder_clowder_1
6126bec23734        elasticsearch:2              "/docker-entrypoint.…"   14 seconds ago      Up 9 seconds                      0.0.0.0:32775->9200/tcp, 0.0.0.0:32774->9300/tcp                                              clowder_elasticsearch_1
483eeefe8a2c        rabbitmq:management-alpine   "docker-entrypoint.s…"   14 seconds ago      Up 10 seconds                     4369/tcp, 5671/tcp, 15671/tcp, 25672/tcp, 0.0.0.0:15672->15672/tcp, 0.0.0.0:32772->5672/tcp   clowder_rabbitmq_1
b70f3daf40de        mongo:3.4                    "docker-entrypoint.s…"   14 seconds ago      Up 7 seconds                      0.0.0.0:32773->27017/tcp                                                                      clowder_mongo_1
```
- docker run -it --rm --network clowder_clowder -e "MONGO_URI=mongodb://clowder:27017/clowder" -e "FIRST_NAME=test" -e "LAST_NAME=test" -e "EMAIL_ADDRESS=test@example.com" -e "USERNAME=test@example.com" -e "PASSWORD=testPassword" -e "ADMIN=false" clowder/mongo-init

Error message:
```Traceback (most recent call last):
  File "/mongo-init.py", line 63, in <module>
    if users.find_one({"identityId.userId": user_name, "identityId.providerId": "userpass"}):
  File "/usr/local/lib/python3.7/site-packages/pymongo/collection.py", line 1269, in find_one
    for result in cursor.limit(-1):
  File "/usr/local/lib/python3.7/site-packages/pymongo/cursor.py", line 1225, in next
    if len(self.__data) or self._refresh():
  File "/usr/local/lib/python3.7/site-packages/pymongo/cursor.py", line 1117, in _refresh
    self.__session = self.__collection.database.client._ensure_session()
  File "/usr/local/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1598, in _ensure_session
    return self.__start_session(True, causal_consistency=False)
  File "/usr/local/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1551, in __start_session
    server_session = self._get_server_session()
  File "/usr/local/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1584, in _get_server_session
    return self._topology.get_server_session()
  File "/usr/local/lib/python3.7/site-packages/pymongo/topology.py", line 429, in get_server_session
    None)
  File "/usr/local/lib/python3.7/site-packages/pymongo/topology.py", line 200, in _select_servers_loop
    self._error_message(selector))
pymongo.errors.ServerSelectionTimeoutError: clowder:27017: [Errno 111] Connection refused
```

Reply:
MONGO_URI should be `mongodb://mongo:27017/clowder`

Reply:
That did the trick! 

I had copied the MONGO_URI from the script portion of the README at the above link and not the command line example. The URL from the first example is what worked

Reply:
Thanks @robkooper for reviewing, your feedback, and your help with this